sprockets/mixins/amqp/__init__.py

```diff
@@ -571,9 +576,10 @@ def on_channel_closed(self, channel, reply_code, reply_text):
         :param str reply_text: The text reason the channel was closed
 
         """
+        LOGGER.debug('Channel is closed')
         for future in self.messages.values():
             future.set_exception(AMQPException(reply_code, reply_text))
-        self.messages = {}
+
```

Comment:
You moved this code into a block that only executes if publisher confirmations are on, is there a case where you'll have messages hang around that shouldn't when that is turned off?

Reply:
`self.messages` was added purely for publisher confirmation bookkeeping, see how it's used here: https://github.com/sprockets/sprockets.mixins.amqp/blob/c12bdf5ceba5447cc21dba92402d24de27f825f1/sprockets/mixins/amqp/__init__.py#L238-L240

So if confirmations is not enabled then it's not used.